### ISSUE TITLE
Mac OSX compile

### ISSUE THREAD

**Issue by phishee** [badaix/snapcast#309]
Has anyone successfully compiled the snapcast client or server on Mac OSX 10.11 Capitan?  I receive the following error when attempt it:

`../externals/aixlog/include/aixlog.hpp:42:10: fatal error: 'os/log.h' file not found`

Any ideas?  What am I missing?  I would love to be able to use snapcast on my mac mini.  
Thanks!

**Comment by ThYpHo0n** [comment]
Should work fine on the current master the OSX build is passing just fine, we’re building against xcode 9.1 though, you can see the working steps to reproduce a build in our TravisCI config and the job log:
https://travis-ci.org/badaix/snapcast/jobs/301092586
Have you checked out all git sub modules correctly (recursively)?

**Comment by badaix** [comment]
Snapcast uses unified logging on Mac OS. According to the Apple developer website you will need at least macOS 10.12
```
Important

Unified logging is available in iOS 10.0 and later, macOS 10.12 and later,  
tvOS 10.0 and later, and watchOS 3.0 and later, and supersedes ASL  
(Apple System Logger) and the Syslog APIs. Historically, log messages  
were written to specific locations on disk, such as /etc/system.log.  
The unified logging system stores messages in memory and in a data store,  
rather than writing to text-based log files.
```
A solution would be a log sink for this ASL (Apple System Logger) in [AixLog](https://github.com/badaix/aixlog). Maybe it would even work to use the existing syslog sink instead. It should be possible to use macros in `Availability.h`, as discussed [here](https://stackoverflow.com/questions/8024961/c-macro-for-os-x-version-lion-or-not-detection), to check the version during compile time and use one log sink or the other.  
Unfortunately, I had to give my lent mac book back just yesterday. No joke :(
